torch/optim/adadelta.py

```diff
@@ -310,8 +310,8 @@ def _single_tensor_adadelta(
     # If compiling, the compiler will handle cudagraph checks, see note [torch.compile x capturable]
     if not torch._utils.is_compiling() and capturable:
         assert all(
-            p.is_cuda and step.is_cuda for p, step in zip(params, state_steps)
-        ), "If capturable=True, params and state_steps must be CUDA tensors."
+            p.device == step.device for p, step in zip(params, state_steps)
```

Comment:
Now that I'm seeing that this is less safe than before, I think we should add a p's device type is in a _supports_capturable_devices list = ["cuda", "xla", "privateuse1"]. The list can live in optimizer.py for now.

Reply:
@janeyx99 
1. I find that some optimizers do not check xla now，is it ok if I add xla in  _supports_capturable_devices?
2. I will change to the below code, is it fine with you?
```
if not torch._utils.is_compiling() and capturable:
        assert all(
            p.device.type == step.devcie.type
            and p.device.type in _supports_capturable_devices_list
            for p, step in zip(params, state_steps)
        ), f"If capturable=True, params and state_steps must be on supported devices: {_supports_capturable_devices_list}."
```